Jesus Branch

jl5211 opened the pull request:
adding FB SDK

jl5211 committed:
alamofire

added

jl5211 committed:
Merge branch 'develop' into Jesus_Branch

# Conflicts:
#	Podfile
#	Podfile.lock

jl5211 committed:
attempting to push

did it work

jl5211 committed:
fbsdk

added the fbsdk, 2 errors, one in RFLocationDetailViewController, and
one in FbLoginViewController

jl5211 committed:
updating from develop

jl5211 committed:
c

rfree18 commented on NYUDining/FbLoginViewController.swift:
```diff
@@ -0,0 +1,54 @@
+    {
+        FBSDKGraphRequest.init(graphPath: "me", parameters: ["fields":"first_name, last_name, picture.type(large)"]).startWithCompletionHandler { (connection, result, error) -> Void in
+            
+            let strFirstName: String = (result.objectForKey("first_name") as? String)!
```
You should never force unwrap an optional, even if you're practically certain it will never be nil. It's just safe practice to prevent unwanted crashes. This could be written as `if let strFirstName = result.objectForKey("first_name") as String`

rfree18 commented on NYUDining/EditProfileViewController.swift:
```diff
@@ -0,0 +1,67 @@
+    
+    @IBOutlet weak var major: UITextField!
+    
+    var schoolDataSource : Array = ["College of Arts & Science", "Graduate School of Arts and Science", "College of Dentistry", "College of Global Public Health", "Courant Institute", "Gallatin School", "Institute of Fine Arts", "Institute for the Study of the Ancient World", "Leonard N. Stern School of Business", "Robert F. Wagner Graduate School of Public Service", "Rory Meyers College of Nursi ...
```
This would be better suited as an enum or in some sort of tag file that quite literally just holds string constants (so for example you'd just call SchoolNames.Stern)

jl5211 committed:
FCM

Added code needed for put notifications with FCM

jl5211 committed:
few edits

set up alomofire connections

rfree18 commented on NYUDining/EditProfileViewController.swift:
```diff
@@ -0,0 +1,143 @@
+    @IBOutlet weak var major: UITextField!
+
+    enum NYUSchools: Int, CustomStringConvertible {
+        case CAS = 0
```
No need to specify numbers, it's inferred from the type

rfree18 commented on NYUDining/EditProfileViewController.swift:
```diff
@@ -0,0 +1,143 @@
+            case .TSA : return "Tisch School of the Arts"
+            case .AD : return "NYU Abu Dhabi"
+            case .SH : return "NYU Shanghai"
+            default : return " "
```
Default shouldn't be needed since you're going through every possible case

rfree18 commented on NYUDining/FbLoginViewController.swift:
```diff
@@ -0,0 +1,86 @@
+            {
+                let fbUserId: String = (result.objectForKey("id") as? String)!
+                print("\(fbUserId)")
+                let strFirstName: String = (result.objectForKey("first_name") as? String)!
```
This logic should are rewritten, probably by using SwiftyJSON. The main problem now is that you're force unwrapping everything. While it isn't a problem assuming the JSON file is always 100% perfect, it's still never a good idea to force unwrap - that's the cause for like 90% of crashes I've experienced.

Commit:
Major changes

Commit:
Visual improvments

Commit:
Profile view

Commit:
Finish profile view

Commit:
Fix hours label

rfree18 commented on NYUDining/EditProfileViewController.swift:
```diff
@@ -0,0 +1,143 @@
+    @IBAction func saveInfo(sender: AnyObject) {
+        print ("save button is pressed")
+        let defaults = NSUserDefaults.standardUserDefaults()
+        defaults.setObject(year.text, forKey: "School_Year")
```
Not a good usage of NSUserDefaults. It would be better to have a flow like this: on initial login, collect user's information (school, major, etc.) and create a new user on the backend with that information. When going to display the information, the app should call the backend to retrieve all of the information. There should be a new User model class/struct that stores this information. Using NSUserDefaults can cause a ton of bugs, especially if it's not cleared properly on logout.

Commit:
Bug fixes

Commit:
Profile changes

Commit:
Updates

jl5211 committed:
Added multiple funtionalities

The functionalities have not been fully tested, upon testing it will be
edited, messaging features has not yet been included

jl5211 committed:
Merge pull request #26 from rfree18/V2.0

V2.0

jl5211 committed:
requestButton

Forgot to add the put request for accepting a request

jl5211 committed:
Merge pull request #29 from rfree18/V2.0

V2.0
